### ISSUE TITLE
What is in a name?

### ISSUE THREAD

**Issue by stappersg** [restic/rest-server#90]
Why do we stick to the name _rest-server_?

And what would be [a better name?](https://duckduckgo.com/?q=rest+server)

**Comment by jooola** [comment]
I'm all for it. What about `restic-server` ?

**Comment by rawtaz** [comment]
@stappersg Because that is exactly what it is - a REST server. Under the restic namespace.

I fail to see what problem you are suggesting needs to be solved by renaming this well established software.

@jooola That name is too generic, unless other types of protocols would be implemented in this software, when needed.

**Comment by stappersg** [comment]
Yes,  the "problem" is about using this _REST server_ in a wider name space as the _restic namespace_.

**Comment by rawtaz** [comment]
@stappersg Can you rephrase that? :) I don't get what you're trying to say with the last sentence you wrote.

**Comment by gbolo** [comment]
I think `restic-server` is more appropriate. Or perhaps `restic-backend`. The current name is too generic

**Comment by rawtaz** [comment]
How do you mean it's too generic when it's specifically about a REST server? That's not generic, that's specific. And unless this server is servicing more than one type of protocol it probably should be.

**Comment by gbolo** [comment]
> How do you mean it's too generic when it's specifically about a REST server? That's not generic, that's specific. And unless this server is servicing more than one type of protocol it probably should be.

@rawtaz from a github repo perspective its fine (`restic/rest-server`), but the resulting binary has the generic name of `rest-server`. Anyways, not a big deal to me, but thought i would add my 2 cents.

**Comment by stappersg** [comment]
> @stappersg Can you rephrase that? :) I don't get what you're trying to say with the last sentence you wrote.


With all respect for creating _restic_ and _rest-server_, but the name _rest-server_ is way tooo generic outside the _restic bubble_.

**Comment by mholt** [comment]
Ideally, you'd never make a `rest-server` binary. I think there should be a way to embed the rest-server into the `restic` command, and then `restic/rest-server` is only ever used as a library, which would resolve the complaints here.

**Comment by jooola** [comment]
@mholt One concern of binding the server inside the restic binary is the release process. Releasing a new version of the server will require a release of a new restic version.

I would be for a name change, from `rest-server` to `restic-server` *at least* for the binary. Migration process can be done using a dummy binary that forward the call to the real binary, with a prepended deprecation notice. They are many ways to do this.

**Comment by rawtaz** [comment]
> > @stappersg Can you rephrase that? :) I don't get what you're trying to say with the last sentence you wrote.
> 
> With all respect for creating _restic_ and _rest-server_, but the name _rest-server_ is way tooo generic outside the _restic bubble_.

So the software world should go around renaming `httpd` to `apache-server`? Gimme a break ;)

There will always be name collisions in the general namespace, it happens all the time with software suites containing multiple binaries for different purposes. If you rename `rest-server` to `restic-server`, what will you then do when (if) we introduce a second protocol for serving a repository in the restic bubble? You'll have to implement it in the same binary as a separate command instead. At *that* point it starts making more sense to rename `rest-server` to a more generic / less specific name, but right now this software is exactly what it's named - a REST server. It's not even very restic specific, it serves files over a REST interface, rather than specifically serving a restic repository. So the name as it is now corresponds better than what `restic-server` would.

I digress, because this is probably the least important thing to focus on in restic and rest-server today, and in the end it's up to @fd0 anyway :)

**Comment by stappersg** [comment]
I tent to go for `restic-rest-server` as name for the Debian package of the _restic_  _rest-server_.

**Comment by rawtaz** [comment]
Thank you everyone for your input, it's all been good. I've decided to close this issue for now. It can be reopened later if need be, but for the time being we'll keep the name as it is. It's recurringly being used as-is and without problems from what I can tell. At the moment there are other things which are more fruitful to spend time on. In the future be might incorporate rest-server into restic itself, I think that's a pretty sane direction to go in down the road. But we'll see.
